### PR TITLE
Join as maintainer + rerender

### PR DESCRIPTION
**Pull request by h-vetinari**
Rebasing #13 without windows. If/once this is merged, I'd then try to catch up with the currently missing releases, and _then_ retry windows again.

### CODE REVIEW

**Comment by h-vetinari** [comment]
PTAL @conda-forge/lmdb @jakirkham

**Comment by h-vetinari** [comment]
Ping @jakirkham 🙃

**Comment by h-vetinari** [comment]
Re-ping @jakirkham (since you're the only maintainer here...)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,19 +21,35 @@ requirements:
...
+    # binaries
     - mdb_copy -V                                    # [unix]
     - mdb_dump -V                                    # [unix]
     - mdb_load -V                                    # [unix]
     - mdb_stat -V                                    # [unix]
-    - test -f ${PREFIX}/lib/liblmdb.a                # [unix]
+
+    # shared:
     - test -f ${PREFIX}/lib/liblmdb.so               # [unix]
+    - if not exist %LIBRARY_BIN%\lmdb.dll exit 1     # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we test for the corresponding `.lib` as well?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,19 +21,35 @@ requirements:
...
+
+    # TODO: do we want to do have static libraries at all?
+    # static:
+    - test -f ${PREFIX}/lib/liblmdb.a                # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: No we should remove this per [CFEP 18]( https://github.com/conda-forge/cfep/blob/2409812d68296fcf2e06255687cc7cb085923ffd/cfep-18.md )

File: recipe/meta.yaml [hunk 3]
```diff
@@ -24,19 +21,35 @@ requirements:
...
-  home: http://symas.com/mdb/
-  license: OpenLDAP Public
+  home: https://www.openldap.org/software/repo.html
+  license: OLDAP-2.8
   license_file: libraries/liblmdb/LICENSE
   summary: A high-performance embedded transactional key-value store database.
+  # careful: the gitlab repo has different branches for very
+  # different projects - relevant here are the ones called mdb.*;
+  # also: there's a mirror under https://github.com/LMDB/lmdb
+  dev_url: https://git.openldap.org/openldap/openldap/-/tree/mdb.master
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we just link to the mirror? It is pretty clear how to get to maintained repos for there and makes it a bit easier to see what is LMDB specific

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,20 +1,17 @@
...
   url: https://github.com/LMDB/lmdb/archive/LMDB_{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: 44602436c52c29d4f301f55f6fd8115f945469b868348e3cddaf91ab2473ea26
   patches:
-    - Makefile.patch
+    - patches/Makefile.patch
 
 build:
-  number: {{ build }}
+  number: 1
```
Comment (jakirkham) [inline recipe/meta.yaml:14]: Why is the build number bump needed? Are we changing something in the package? Looks mostly like style and metadata changes

**Review by jakirkham — COMMENTED** [review]
Thanks Axel 😄

Had a few comments below

File: recipe/meta.yaml (hunk 4 above)
Comment (h-vetinari) [reply recipe/meta.yaml:14]: I tend to be careful when the infrastructure changes, which in this case, is a lot 😅 
```diff
 c_compiler_version:
-- '4'
+- '12
```

File: recipe/meta.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/meta.yaml]: I don't claim to understand all of the windows specifics, but it is my understanding that not every DLL _needs_ a `.lib`? I'd cross-check the logs, but then, we're not building windows yet. I plan to march through the existing [tags](https://github.com/LMDB/lmdb/tags) before attempting that.

File: recipe/meta.yaml (hunk 3 above)
Comment (h-vetinari) [reply recipe/meta.yaml]: Fine by me.

File: recipe/meta.yaml (hunk 2 above)
Comment (h-vetinari) [reply recipe/meta.yaml]: OK.

**Review by h-vetinari — COMMENTED** [review]
Thanks for taking a look!

File: recipe/meta.yaml (hunk 4 above)
Comment (jakirkham) [reply recipe/meta.yaml:14]: Ah good point!

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml]: IIUC we do need it for [dynamic linking on Windows]( https://docs.microsoft.com/en-us/troubleshoot/windows-client/deployment/dynamic-link-library#load-time-dynamic-linking )

That said, given we are not attempting building on Windows yet, maybe we can leave out the Windows changes until we intend to add it

Comment (h-vetinari) [reply recipe/meta.yaml]: I don't see the issue with two extra lines of tests, but if you say so...

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by jakirkham** [comment]
Thanks Axel! 😄

**Pull request merged**